src/cryptography/x509/base.py

```diff
@@ -1169,5 +1184,69 @@ def build(self, backend: typing.Any = None) -> RevokedCertificate:
         )
 
 
+class Profile(utils.Enum):
+    RFC5280 = 0
+    WebPKI = 1
+
+
+class PolicyBuilder:
+    def __init__(
+        self,
+        *,
+        name: typing.Optional[str] = None,
+        time: typing.Optional[datetime.datetime] = None,
+        profile: Profile = Profile.WebPKI,
+    ):
+        self._name = name
+        self._time = time
+        self._profile = profile
+
+    def name(self, new_name: str) -> PolicyBuilder:
```

Comment:
I don't think `str` is sufficient for expected name -- how does it distinguish between a DNSName and an IP Address?

Reply:
Yeah, it isn't -- this is just a sketch. I think this will need to `DNSName | IPAddress` or similar.